src/h5web/providers/hsds/utils.ts

```diff
@@ -52,11 +52,15 @@ export function convertHsdsBaseType(hsdsBaseType: HsdsBaseType): HDF5BaseType {
 
   const [, sign, size, endianness] = matches;
 
-  return {
+  const hdf5Type = {
     class: hsdsClass,
     endianness: endianness as HDF5Endianness,
     size: Number.parseInt(size, 10),
-    unsigned: sign === 'U' || undefined,
+  };
+
+  return {
+    ...hdf5Type,
+    ...(sign === 'U' ? { unsigned: true } : {}),
   };
```

Comment:
oh, no need for the separate return here 😉

Reply:
:facepalm: thanks